pages/examples/features/authentication/modal.tsx

```diff
@@ -7,37 +7,36 @@ import Button from '@system/Button';
 import Content from '@system/layouts/Content';
 import Footer from '@system/Footer';
 import GlobalModalManager from '@system/modals/GlobalModalManager';
+import ModalAuthentication from '@demos/modals/ModalAuthentication';
 import Navigation from '@system/Navigation';
 import Page from '@components/Page';
 import SectionFullHeight from '@system/sections/SectionFullHeight';
 
-import { useModal } from '@system/providers/ModalContextProvider';
+import { useModal } from '@root/system/modals/ModalContext';
 
 function ExampleModals(props) {
-  const { showModal } = useModal();
+  const modalAuthentication = useModal(ModalAuthentication);
```

Comment:
What I don't like about this approach is what about having multiple modals, so if I wanted to have one modal on my screen, and then switch to another modal?

Reply:
do you mean you want to have two modals open at once?

currently, upon opening a new modal, any other open modal gets closed.

my understanding of a modal system is that they are generally mutually exclusive, with multiple modals being the exception.

the solution that i planned out for this is to add a layer tagging system. when opening a modal, there would be an optional parameter for a string representing the layer to place the modal on. two modals that are on different layers could then be open at the same time.

i hope the answer you were looking for is somewhere in this response, although honestly i didn't really fully understand the question. happy to follow up.

Reply:
When you write code like this, you are suggesting the right pattern to follow is

```
const modalA = useModal(ModalA);
const modalB = useModal(ModalB);
const modalC = useModal(ModalC);
```

Or it gets to the point where it is like

```
const modals = useModals([ModalA, ModalB, ModalC]);
```

And I'm not sure I like either... 

The original intent was to pass a modals singleton that could handle various modal operations like `swap`, `render`, and `hide`. I don’t think I communicated this clearly while you were working on the modals, but I was hoping you’d arrive at a similar approach by looking at the code. Essentially, anything related to rendering and managing the modal should be part of this singleton.

Reply:
ok, i get it now. i can change this. the way i did it had been in consideration of strong typing, but thinking about it again now that the system is all put together, i can see that it's not necessary to have separate modal handles. i'll consolidate them into a single one.

Reply:
also, there is no swap function. there is just an open function, because opening a modal causes the previous one to be closed.

i built the system so that we can extend it later to allow modal components to define a `canClose()` function, so that modals can prevent themselves from being closed / swapped out e.g. if there are unsaved changes.

Reply:
I think you get it now though that its more like a `ModalsService` that brokers and handles modals. So I trust you to architect it correctly over time.

Reply:
applied your suggestion